apps/api-documenter/src/JsonFilePaths.ts

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+declare const __webpack_require__: unknown;
+
+export class JsonFilePaths {
+  public static apiDocumenterSchemaPath: string =
+    typeof __webpack_require__ !== 'undefined'
+      ? require('file-loader?name=[name]_[contenthash:8].[ext]!./schemas/api-documenter.schema.json')
+      : path.join(__dirname, 'schemas', 'api-documenter.schema.json');
```

Comment:
👎 This looks pretty ugly.

Webpack's directives like `file-loader?name=[name]_[contenthash:8].[ext]!` are a poor developer experience.  The toolchain's bundling strategy should be specified once, in a rig config file.  It should not be repeatedly specified for each `.schema.json` file.  And it should definitely not be specified inside each import statement in a source file.  (I get that this is a "recommended" practice, but IMO Webpack doesn't have a reliable barometer for what should be recommended. 😋.)

Fundamentally, the original code used `__dirname` to locate files on disk relative to the `JsonFilePaths.ts` source file.  This is incompatible with bundling, because `JsonFilePaths.ts` doesn't have a meaningful file path inside the bundle.  We need to actually solve that.

Either:

1. We should continue loading `api-documenter.schema.json` from a file on disk, and simply replace `__dirname` with a different mechanism that returns a meaningful path.  Maybe introduce a utility like `getPackageJsonFolder()` or something.

   -- OR --

2. We should reinterpret `api-documenter.schema.json` as a JavaScript module that gets imported and bundled.  In this case `path.join(__dirname, 'schemas', 'api-documenter.schema.json');` should be completely replaced by a conventional `import` statement. Any relevant webpack configuration should be applied generically based on the `.json` file extension. BTW TypeScript directly supports JSON file imports now.

#<!-- -->1 definitely seems like the better solution, since the "schemas" folder is meant to be visible to external consumers of the package.